application/comit_node/src/http_api/rfc003/swap.rs

```diff
@@ -273,22 +213,152 @@ fn handle_post_swap<A: AliceSpawner>(
 }
 
 #[derive(Debug, Serialize)]
-pub struct SwapDescription {
+struct GetSwapResource {
+    pub swap: SwapDescription,
+    pub role: String,
+    pub state: SwapState,
+}
+
+#[derive(Debug, Serialize)]
+struct SwapDescription {
     alpha_ledger: HttpLedger,
     beta_ledger: HttpLedger,
     alpha_asset: HttpAsset,
     beta_asset: HttpAsset,
+}
+
+#[derive(Debug, Serialize)]
+struct SwapState {
+    outcome: SwapOutcome,
+    communication: SwapCommunication,
+    alpha_ledger: LedgerState,
+    beta_ledger: LedgerState,
+}
+
+// TODO: SwapOutcome::InternalFailure is not yet supported
+#[derive(Debug, Serialize)]
+enum SwapOutcome {
+    InProgress,
+    Swapped,
+    NotSwapped,
+}
+
+#[derive(Debug, Serialize)]
+struct SwapCommunication {
+    current_state: SwapCommunicationState,
     alpha_expiry: Timestamp,
     beta_expiry: Timestamp,
+    alpha_redeem_identity: Option<Value>,
```

Comment:
:question: Should the identities really be a `Value` here?

Reply:
I thought about using `Ledger::Identity`s, but that would mean passing type parameters to some of these HTTP structs. Would that be preferable?